src/main/java/com/deepoove/swagger/diff/compare/ParameterDiff.java

```diff
@@ -24,41 +24,37 @@
   private List<Parameter> missing;
   private List<ChangedParameter> changed;
 
-  Map<String, Model> oldDedinitions;
-  Map<String, Model> newDedinitions;
+  private Map<String, Model> oldDefinitions;
+  private Map<String, Model> newDefinitions;
 
-  private ParameterDiff() {
-  }
+  private boolean hasOnlyCosmeticChanges;
 
-  public static ParameterDiff buildWithDefinition(Map<String, Model> left,
-                                                  Map<String, Model> right) {
-    ParameterDiff diff = new ParameterDiff();
-    diff.oldDedinitions = left;
-    diff.newDedinitions = right;
-    return diff;
+  public ParameterDiff(Map<String, Model> left, Map<String, Model> right) {
+    this.oldDefinitions = left;
+    this.newDefinitions = right;
+    this.hasOnlyCosmeticChanges = false;
   }
 
-  public ParameterDiff diff(List<Parameter> left,
-                            List<Parameter> right) {
-    ParameterDiff instance = new ParameterDiff();
+  public void diff(List<Parameter> left,
```

Comment:
You chose to change the signature for a lot of the functions from an object to `void`, and instead have those results be available via getters. What's your reasoning for those changes?

Reply:
The signatures I changed do not have getters for what they used to return, because these `compare` and `diff` methods were called on instances, but they made new instances and returned those instead. I thought that was bad OOP practice of not using constructors and member functions properly, especially when a function is being called on an instance but a new class instance is being created and returned.

Reply:
What was really happening here was the conflation between a diff-operator's configuration and its resulting data. In most cases, the existing API is basically currying the configuration data when it constructs an instance.

If we're going to change the interface this much, I'd prefer that we just move the resulting data into separate models (ie `ParameterDiffResult`). Then the instantiation of a diff object would really be configuring it for performing diffs between the models passed into its methods.

I think using side effects for everything makes it much harder to grok the current state of the world at any point, which could lead to difficulties debugging these classes (which can already be confusing to follow things through).

Reply:
Are you suggesting separating the existing diff classes (like `ParameterDiff`) into two classes - a `...Diff` class that stores the objects that contain the things being compared (something like `definitions` for `ParameterDiff`) and a `...DiffResult` class that stores the objects that track the changes (like `increased...` and `missing...` lists etc.)? The `...Diff` classes would also contain the actual diff logic, while the `...DiffResult` classes would pretty much just be data classes. I would agree with this, but maybe I'm misunderstanding?

Reply:
Yep, that's exactly what I mean. I think we should trend towards the better separation and avoiding the side effect implementation here